excel/src/main/java/org/apache/metamodel/excel/DefaultSpreadsheetReaderDelegate.java

```diff
@@ -167,50 +166,114 @@ private MutableTable createTable(final Workbook wb, final Sheet sheet) {
                     break;
                 }
             }
-
             if (hasColumns) {
-                createColumns(table, wb, row);
+                createColumns(table, wb, row, columnTypes);
             }
         }
-
         return table;
     }
 
+    private void setColumnType(Iterator<Row> data, int rowLength, ColumnType[] columnTypes) {
+        while (data.hasNext()) {
```

Comment:
I had a query to look up only the first 1000records, I see that that got removed when getting DataRow types instead of the "normal" Row type.
I've added a countdown from 1000 in the while-loop
```
int eagerness = 1000;
...
...
while(data.hasnext() && eagerness-- > 0) {
...
...
```